app/views/handlers/confirm_email.py

```diff
@@ -153,17 +148,17 @@ def _publish_fulfilment_request(self):
 @dataclass
 class ConfirmationEmailFulfilmentRequest(FulfilmentRequest):
     email_address: str
-    session_data: SessionData
+    metadata: dict[str, Any]
     schema: QuestionnaireSchema
 
     def _payload(self) -> Mapping:
         return {
             "fulfilmentRequest": {
                 "email_address": self.email_address,
-                "display_address": self.session_data.display_address,
+                "display_address": self.metadata["display_address"],
                 "form_type": self.schema.form_type,
-                "language_code": self.session_data.language_code,
+                "language_code": self.metadata["language_code"],
```

Comment:
This language code should be the one from session as it should be the language the user is currently in.